src/app/Rules/MatchExpressionEvaluator.tsx

```diff
@@ -0,0 +1,141 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import { CodeBlock, CodeBlockCode, Label, Stack, StackItem, Text, ValidatedOptions } from '@patternfly/react-core';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import { Target } from '@app/Shared/Services/Target.service';
+import { TargetSelect } from '@app/TargetSelect/TargetSelect';
+import { NoTargetSelected } from '@app/TargetView/NoTargetSelected';
+import {CheckCircleIcon, InfoCircleIcon, QuestionCircleIcon, WarningTriangleIcon} from '@patternfly/react-icons';
+
+export interface MatchExpressionEvaluatorProps {
+  matchExpression?: string;
+}
+
+export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEvaluatorProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [target, setTarget] = React.useState(undefined as Target | undefined);
+  const [valid, setValid] = React.useState(ValidatedOptions.default);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target.target().subscribe(setTarget)
+    );
+  }, [context, context.target]);
+
+  React.useEffect(() => {
+    if (!props.matchExpression || !target?.connectUrl) {
+      setValid(ValidatedOptions.default);
+      return;
+    }
+    try {
+      const f = new Function('target', `return ${props.matchExpression}`);
+      const res = f(target);
+      if (typeof res === 'boolean') {
+        setValid(res ? ValidatedOptions.success : ValidatedOptions.warning);
+        return;
+      }
+      setValid(ValidatedOptions.error);
+      return;
+    } catch (err) {
+      setValid(ValidatedOptions.error);
+      return;
+    }
+  }, [target, props.matchExpression]);
+
+  const statusLabel = React.useMemo(() => {
+    switch (valid) {
+      case ValidatedOptions.success:
+        return (<Label color="green" icon={<CheckCircleIcon />}>Expression Matches Selected Target</Label>);
+      case ValidatedOptions.warning:
+        return (<Label color="orange" icon={<QuestionCircleIcon />}>Expression Valid, No Match</Label>);
```

Comment:
For the 'Expression Valid, No Match' label, I originally interpreted this as 'the expression is valid and does not match any targets'. Maybe you could reword it as 'Expression Valid, Does Not Match Selected Target' or something similar?